src/screens/SearchScreen.js

```diff
@@ -0,0 +1,175 @@
+import React, {Component} from 'react';
+import {View, Text, ActivityIndicator, StyleSheet, Platform, Keyboard} from 'react-native';
+import {Button} from 'react-native-elements';
+import {MapView} from 'expo';
+import {Icon} from 'react-native-elements';
+
+import _ from 'lodash';
+
+import {CardSection, Card} from '../components/common/index';
+
+const SearchOptions = {
+    MAP: 'map',
+    LIST: 'list'
+};
+
+
+class SearchScreen extends Component {
+    state = {
+        region: {
+            longitude: -105.727939,
+            latitude: 39.695168,
+            longitudeDelta: 1,
+            latitudeDelta: 1
+        },
+        mapLoaded: false,
+        viewStyle: SearchOptions.MAP
+    };
+
+    toggleButton = () => {
+        const {viewStyle} = this.state;
+
+        const nextButtonName = viewStyle;
+        const newViewStyle = viewStyle === SearchOptions.MAP ? SearchOptions.LIST : SearchOptions.MAP;
+
+        this.props.navigation.setParams({buttonName: nextButtonName});
+        this.setState({viewStyle: newViewStyle});
+    };
```

Comment:
This `toggleButton` function is passed to the `static` method below. Through closures, it has access to `this` where the `static` method would not. This is the tricky workaround to get the `static` class to be able to modify our state in-flight.